docs/excel/custom-functions-runtime.md

```diff
@@ -0,0 +1,149 @@
+}
+```
+
+## AsyncStorage
+
+AsyncStorage is a key-value storage system that can be used to store authentication tokens or used for:
+
+- Persistent
+- Unencrypted
+- Asynchronous
+
+Additionally, AsyncStorage is available globally to all parts of your add-in. For custom functions, Async Storage is exposed as a global object. For other parts of your add-in, such as taskpanes and other elements which utilize the typical WebView runtime, AsyncStorage is exposed through `Office.Runtime`.
+
+ Methods available on AsyncStorage include getItem, setItem, removeItem, clear, getAllKeys, flushGetRequests, multiGet, multiSet, and multiRemove. At this time, mergeItem and multiMerge are not supported methods.
+
+Each add-in has its own storage partition, with a default of 5MB of storage.  
+
+The code sample below illustrates the process of getting an item from AsyncStorage:
+
+```js
+_goGetData = async () => {
+    try {
+        const value = await AsyncStorage.getItem('toDoItem');
+        if (value !== null) {
+            //data exists and you can do something with it here
+            }
+        } catch (error) {
+            //handle errors here
+        }
+    }
+}
+```
+
+## Dialog API
+
+The Dialog API allows you to require user authentication through an outside resource, such as Google or Facebook, before they can use your function. The Dialog API allows you to open a dialog box which prompts user sign-in.  
+
+The code sample below illustrates the use of the Dialog API’s `displayWebDialog()` method.  
+
+```ts
+// Get auth token before calling my service, a hypothetical API which will deliver a stock price based on stock ticker string, such as "MSFT"
+async function getStock(ticker) {
+    const token = await getToken();
+    let data = await (await fetch(https://myservice.com/?token=token&ticker= + ticker).json());
+    return data.price;
+}
+
+async function getToken() {
+    if (_cachedToken) {
+        return _cachedToken;
+    } else {
+        return await getTokenViaDialog_AsPromise();
+    }
+}
+  
+// Function to display dialog window
+function getTokenViaDialog_AsPromise() {
+    return new Promise ((resolve, reject) => {
+        displayWebDialog("https://www.auth.com/", {
+           height: ’50’,
+           width: ’50%’,
+           hideTitle: true,
+           onMessage: (message, dialog) => {
+                let json = JSON.parse(message);
+                    if (json.type === "token_succeeded") {
+                        resolve(json.value);
+                        dialog.closeDialog();
+                        return;
+                    }
+            // Otherwise, handle other messages.
+           },
+           onClose: () => reject("User closed dialog"),
+        }).catch(e => reject(e));
+    });
+}
+```
+
+> [!NOTE]
+> The Dialog API in the new JavaScript runtime differs from the [current Dialog API](../develop/dialog-api-in-office-add-ins.md) which works for add-ins which are not custom functions.  
```

Comment:
This note is confusing as written; suggest rewording for clarity.  Perhaps something like:

The Dialog API described in this section is part of the new JavaScript runtime for custom functions. This API is different from the [Dialog API](../develop/dialog-api-in-office-add-ins.md) that can be used by add-ins that do not contain custom functions.

Reply:
@mscharlock @kbrandl Actually, an add-in can contain both custom functions and a task pane. So the distinction between the new and old Dialog APIs is in where you use it. The old is used in task panes and add-in commands. The new is used in custom functions.

Reply:
Thanks, I had been struggling with a good way to put it.